### DISCUSSION TITLE
Define timestamp format for JsonEncoder

### DISCUSSION THREAD

**Discussion by YarochkinMichael** [qos-ch/logback#698] [Ideas]
Hello

I am using Fluend sidecar to send logs to ElasticSearch.
Sending timestamp in unix timestamp is not human readable.
Is it possible to set the timestamp format to use?

Something like (line 120)
        appenderMember(sb, TIMESTAMP_ATTR_NAME, new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format( event.getTimeStamp()));

With format setter, of course.

**Comment by YarochkinMichael** [comment]
I have working version in a local branch

**Reply by ceki** [reply]
@YarochkinMichael Can you please describe when does the reading of the timestamp by a human intervene is the process you mention above?

**Reply by YarochkinMichael** [reply]
Unix timestamp is not clear when reading logs on ES.
We even put it as a rule in Fluend configuration. 


`        <parse>
            @type json
            time_type string
            time_key timestamp
            time_format %iso8601
            keep_time_key true
        </parse>
`

**Reply by YarochkinMichael** [reply]
It will be a general improvement to allow customization of logs.

Maybe even make an interface/abstract class and do the same for every encoder.

If not, can we allow injecting FieldCusomizer with spring Beans?

I can prepare a pull request.
